src/Controls/samples/Controls.Sample.UITests/Issues/Issues16561.xaml.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Maui.Controls;
+using Microsoft.Maui.Controls.Xaml;
+using Microsoft.Maui.Platform;
+
+namespace Maui.Controls.Sample.Issues
+{
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+	[Issue(IssueTracker.Github, 16561, "Quick single taps on Android have wrong second tap location", PlatformAffected.Android)]
+	public partial class Issue16561 : ContentPage
+	{
+		int _taps;
+
+		public Issue16561()
+		{
+			InitializeComponent();
+			
+			var tapGesture = new TapGestureRecognizer();
+			tapGesture.Tapped += TapHandler;
+
+			TapArea.GestureRecognizers.Add(tapGesture);
+		}
+
+		void TapHandler(object sender, TappedEventArgs e)
+		{
+			var pos = e.GetPosition(TapArea);
+
+			if (pos == null)
+			{
+				Tap1Label.Text = $"Error, could not get tap position";
+				return;
+			}
+
+#if ANDROID
+			// Adjust the results for display density, so they make sense to the 
+			// Appium test consuming this.
+			var x = this.Handler.MauiContext.Context.ToPixels(pos.Value.X);
+			var y = this.Handler.MauiContext.Context.ToPixels(pos.Value.Y);
+			pos = new(x, y);
+#endif
```

Comment:
The Appium layer doesn't know anything about our SDK's point->pixel conversions (why would it?). So when we're asking it to retrieve the `Rect` for a particular element and then to tap at specific points on that element, it's doing all of that work in native points. The cross-platform tap events are in device-independent units. So we have to do that conversion somewhere. It's easier to do that on the app side, because the app already knows the display density and how to convert the values.

At the moment, we don't have any other UI tests which require this sort of point-to-point comparison. _If_ we start to need more tests like that in the future, we _could_ introduce an automatic conversion mechanism for the Appium `Rect` structure which determines the remote device's display density at startup. But that's not top of my list right now for the one test which has this issue.